README.md

```diff
@@ -1,38 +1,145 @@
+                        Object.prototype.toString.call(workerFilterValue) ===
+                        '[object Array]'
+                    )
+                        //    if at least one selected filter's value matches worker's filter value - locations
+                        return selectedFilterVal.some((filterValue) =>
+                            workerFilterValue.includes(filterValue)
+                        );
+                    // workerFilterValue is a string - contract type and position
+                    else
+                        return selectedFilterVal.some(
+                            (filterValue) => filterValue === workerFilterValue
+                        );
+                });
+            })
+            .map((filteredWorker) => {
+                return `${filteredWorker.imie} ${filteredWorker.nazwisko}`;
+            });
 
-## Installation
+        setFilteredWorkers(selected);
+    };
+    
+    
+    // ja bym to rozbił tak (trochę na szybko, więc niektóre rzeczy pominąłem)
+    const workerFilterValueMatch = () => {}
+    
+    const matchWorkersWithFilters = (defaultFilters, selectedFilters, workers) => {
+        return defaultFilters.every((filter) => {
+            const workerFilterValue = worker[filter];
+            const selectedFilterVal = selectedFilters[filter];
 
-Clone this repository and install dependencies inside project directory.
+            if (workerFilterValueMatch(workerFilterValue)) {
+                return selectedFilterVal.some((filterValue) =>
+                    workerFilterValue.includes(filterValue)
+                );
+            }
+            return selectedFilterVal.some(filterValue => filterValue === workerFilterValue);
+        });
+    }
+    
+    const getFilteredWorkers = (defaultFilters, selectedFilters, workers) => {
+        const selected = workers.reduce(
+            (array, worker) => {
+                const hasMatch = matchWorkersWithFilters(defaultFilters, selectedFilters, workers);
+                if (hasMatch) {
+                    return [
+                        ...array,
+                        `${worker.imie} ${worker.nazwisko}`
+                    ]
+                }
+                return array;
+            }, [])
+
+        setFilteredWorkers(selected);
+    };
 
-```
-git clone git@github.com:justFlow5/employee-filter-app.git
-cd employee-filter-app
-npm install
 ```
 
-## Usage
+Pytania:
+- Na przykładzie home albo text-field - poupychane są styled components? 
+- Czasem używane są do definiowania funkcji function a czasem () => , czy jest za tym jakaś logika
 
-To run this project in development mode use:
+- plik TextField.js 
 
+```javascript=
+    const [isActive, setIsActive] = useState('');
+    const [isFilled, setIsFilled] = useState('');
 ```
-npm start
+Czemu tutaj jest string zamiast np. boolean.
+Rozumiem, że niżej jest to używane jako klasa css , ale np clickedOutside jest już boolean i warunkowe wyświetlenie klasy?
```

Comment:
Tak, isFilled oraz isActive działają jako klasy, a clickedOutside jako warunkowe wyświetlenie klasy - bardzo nieczytelna część mojego kodu, którą dostrzegłem dopiero po tym, jak mi to pokazałeś.